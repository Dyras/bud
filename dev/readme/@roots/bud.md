## Installation

Install [@roots/bud](/packages/@roots/bud) and [**@roots/bud-cli**](/packages/@roots/bud-cli) to your project

```sh
yarn add @roots/bud @roots/bud-cli --dev
```

## Getting started

Bud can either be configured with a static config file (`json`/`yml`) or a builder module (`js`/`ts`).

Dead simple example:

```js
/**
 * bud.config.js
 */
module.exports = (bud) => bud.entry('app', ['app.js']);
```

Or, as yml:

```yml
# bud.config.yml
entry:
  app: 'app.js'
```

A more advanced configuration might look like:

```ts
/**
 * bud.config.ts
 */
import { Framework } from "@roots/bud";

export default (bud: Framework) =>
  bud
    .use([
      require('@roots/bud-babel'),
      require('@roots/bud-postcss'),
      require('@roots/bud-react'),
    ])
    .setPath('src', 'js')
    .setPublicPath('assets')
    .alias({
      '@src': bud.path('src'),
    })

    .entry({ app: 'app.{js,css}' })

    .when(
      bud.isProduction,
      ({ minimize }) => minimize(),
      ({ devtool }) => devtool()
    );
```

For more on configuring [**@roots/bud**](/packages/@roots/bud) check out the [dedicated documentation](/docs/config/README.md).

## Running a build

![Terminal usage](https://github.com/roots/bud/tree/stable/static/img/cli.svg)

Once you've set up your configuration file the following command will run the build:

```sh
yarn bud build
```

You should see your built assets in the `dist` directory of your project.

### Running in `production` mode

```sh
yarn bud build:production
```

### Running in `development` mode

```sh
yarn bud build:development
```

## Example implementations

There are [example implementations available](/examples).

## Extending

Bud provides an intentionally sparse set of out-of-the-box features.

In fact, much of the core of Bud is actually made up of extensions. This is to make it easy for devs to swap out parts of the framework as needed.

That said, you will likely want to utilize extensions in your project.

If you're unsure where to start or what you need you can try the [**@roots/bud-preset-recommend**](/packages/@roots/bud-preset-recommend) preset. Most require zero configuration.

### First-party extensions

There are a number of Roots maintained extensions available to kickstart your project

| Name                                                                                                                         | Project home                                                                                                                                       | Extension docs                                                                                             | Latest                                                                                                       |
| ---------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
| [**@roots/bud-babel**](/packages/@roots/bud-babel)                                   | [**@babel/babel**](https://github.com/babel/babel)                                                                                                 | [ðŸ“š README](/packages/@roots/bud-babel/README.md)                  | ![npm](https://img.shields.io/npm/v/@roots/bud-babel.svg?color=%23525ddc&style=flat-square)                  |
| [**@roots/bud-compress**](/packages/@roots/bud-compress)                             | [**@webpack-contrib/compression-webpack-plugin**](https://github.com/webpack-contrib/compression-webpack-plugin)                                   | [ðŸ“š README](/packages/@roots/bud-compress/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-compress.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-criticalcss**](/packages/@roots/bud-criticalcss)                       | [**@addyosmani/critical**](https://github.com/addyosmani/critical)                                                                                 | [ðŸ“š README](/packages/@roots/bud-criticalcss/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-criticalcss.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-emotion**](/packages/@roots/bud-emotion)                               | [**@emotion/emotion-css**](https://github.com/emotion/emotion-css)                                                                                 | [ðŸ“š README](/packages/@roots/bud-emotion/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-emotion.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-entrypoints**](/packages/@roots/bud-entrypoints)                       | [**@roots/entrypoints-webpack-plugin**](/packages/@roots/entrypoints-webpack-plugin)                       | [ðŸ“š README](/packages/@roots/bud-entrypoints/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-entrypoints.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-esbuild**](/packages/@roots/bud-esbuild)                               | [**@roots/esbuild-loader**](https://github.com/roots/esbuild-loader)                                                                               | [ðŸ“š README](/packages/@roots/bud-esbuild/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-esbuild.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-eslint**](/packages/@roots/bud-eslint)                                 | [**@webpack-contrib/eslint-webpack-plugin**](https://github.com/webpack-contrib/eslint-webpack-plugin)                                             | [ðŸ“š README](/packages/@roots/bud-eslint/README.md)                 | ![npm](https://img.shields.io/npm/v/@roots/bud-eslint.svg?color=%23525ddc&style=flat-square)                 |
| [**@roots/bud-imagemin**](/packages/@roots/bud-imagemin)                             | [**@webpack-contrib/image-minimizer-webpack-plugin**](https://github.com/webpack-contrib/image-minimizer-webpack-plugin)                           | [ðŸ“š README](/packages/@roots/bud-imagemin/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-imagemin.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-library**](/packages/@roots/bud-library)                               | [**@asfktz/autodll-webpack-plugin**](https://github.com/asfktz/autodll-webpack-plugin)                                                             | [ðŸ“š README](/packages/@roots/bud-library/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-library.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-mdx**](/packages/@roots/bud-mdx)                                       | [**@mdx-js/mdx**](https://github.com/mdx-js/mdx)                                                                                                   | [ðŸ“š README](/packages/@roots/bud-mdx/README.md)                    | ![npm](https://img.shields.io/npm/v/@roots/bud-mdx.svg?color=%23525ddc&style=flat-square)                    |
| [**@roots/bud-postcss**](/packages/@roots/bud-postcss)                               | [**@postcss/postcss**](https://github.com/postcss/postcss)                                                                                         | [ðŸ“š README](/packages/@roots/bud-postcss/README.md)                | ![npm](https://img.shields.io/npm/v/@roots/bud-postcss.svg?color=%23525ddc&style=flat-square)                |
| [**@roots/bud-prettier**](/packages/@roots/bud-prettier)                             | [**@prettier/prettier**](https://github.com/prettier/prettier)                                                                                     | [ðŸ“š README](/packages/@roots/bud-prettier/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-prettier.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-purgecss**](/packages/@roots/bud-purgecss)                             | [**@FullHuman/purgecss**](https://github.com/FullHuman/purgecss)                                                                                   | [ðŸ“š README](/packages/@roots/bud-purgecss/README.md)               | ![npm](https://img.shields.io/npm/v/@roots/bud-purgecss.svg?color=%23525ddc&style=flat-square)               |
| [**@roots/bud-react**](/packages/@roots/bud-react)                                   | [**@facebook/react**](https://github.com/facebook/react)                                                                                           | [ðŸ“š README](/packages/@roots/bud-react/README.md)                  | ![npm](https://img.shields.io/npm/v/@roots/bud-react.svg?color=%23525ddc&style=flat-square)                  |
| [**@roots/bud-sass**](/packages/@roots/bud-sass)                                     | [**@sass/sass**](https://github.com/sass/sass)                                                                                                     | [ðŸ“š README](/packages/@roots/bud-sass/README.md)                   | ![npm](https://img.shields.io/npm/v/@roots/bud-sass.svg?color=%23525ddc&style=flat-square)                   |
| [**@roots/bud-stylelint**](/packages/@roots/bud-stylelint)                           | [**@stylelint/stylelint**](https://github.com/stylelint/stylelint)                                                                                 | [ðŸ“š README](/packages/@roots/bud-stylelint/README.md)              | ![npm](https://img.shields.io/npm/v/@roots/bud-stylelint.svg?color=%23525ddc&style=flat-square)              |
| [**@roots/bud-tailwindcss**](/packages/@roots/bud-tailwindcss)                       | [**@tailwindlabs/tailwindcss**](https://github.com/tailwindlabs/tailwindcss)                                                                       | [ðŸ“š README](/packages/@roots/bud-tailwindcss/README.md)            | ![npm](https://img.shields.io/npm/v/@roots/bud-tailwindcss.svg?color=%23525ddc&style=flat-square)            |
| [**@roots/bud-terser**](/packages/@roots/bud-terser)                                 | [**@terser/terser**](https://github.com/terser/terser)                                                                                             | [ðŸ“š README](/packages/@roots/bud-terser/README.md)                 | ![npm](https://img.shields.io/npm/v/@roots/bud-terser.svg?color=%23525ddc&style=flat-square)                 |
| [**@roots/bud-typescript**](/packages/@roots/bud-typescript)                         | [**@TypeStrong/ts-loader**](https://github.com/TypeStrong/ts-loader)                                                                               | [ðŸ“š README](/packages/@roots/bud-typescript/README.md)             | ![npm](https://img.shields.io/npm/v/@roots/bud-typescript.svg?color=%23525ddc&style=flat-square)             |
| [**@roots/bud-vue**](/packages/@roots/bud-vue)                                       | [**@vue/vue**](https://github.com/vue/vue)                                                                                                         | [ðŸ“š README](/packages/@roots/bud-vue/README.md)                    | ![npm](https://img.shields.io/npm/v/@roots/bud-vue.svg?color=%23525ddc&style=flat-square)                    |
| [**@roots/bud-wordpress-dependencies**](/packages/@roots/bud-wordpress-dependencies) | [**@roots/wordpress-dependencies-webpack-plugin**](/packages/@roots/wordpress-dependencies-webpack-plugin) | [ðŸ“š README](/packages/@roots/bud-wordpress-dependencies/README.md) | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-dependencies.svg?color=%23525ddc&style=flat-square) |
| [**@roots/bud-wordpress-externals**](/packages/@roots/bud-wordpress-externals)       | [**@roots/wordpress-externals-webpack-plugin**](/packages/@roots/wordpress-externals-webpack-plugin)       | [ðŸ“š README](/packages/@roots/bud-wordpress-externals/README.md)    | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-externals.svg?color=%23525ddc&style=flat-square)    |
| [**@roots/bud-wordpress-manifests**](/packages/@roots/bud-wordpress-manifests)       | [**@roots/merged-manifest-webpack-plugin**](/packages/@roots/merged-manifest-webpack-plugin)               | [ðŸ“š README](/packages/@roots/bud-wordpress-manifests/README.md)    | ![npm](https://img.shields.io/npm/v/@roots/bud-wordpress-manifests.svg?color=%23525ddc&style=flat-square)    |

### Third-party extensions

Have you produced a Bud extension and want to share it here? Please, create an issue sharing information about your project.
