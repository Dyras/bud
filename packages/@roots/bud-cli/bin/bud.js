#!/usr/bin/env node

const CLI = require('../lib/cjs/CLI').CLI
const commands = require('../lib/cjs/CLI/commands')

new CLI({
  command: 'bud',
  projectUrl: 'https://github.com/roots/bud',
  commands,
})
  .mast()
  .invoke()
