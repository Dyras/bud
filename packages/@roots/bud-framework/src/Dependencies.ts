import {Service} from './Service'

/**
 * Dependencies service container
 *
 * @public @core @container
 */
export interface Dependencies extends Service {
  /**
   * Record of installed packages
   *
   * @public
   */
  installed: Array<[string, string]>

  /**
   * Dependency manager
   *
   * @public
   */
  manager: any

  /**
   * Package manager
   *
   * @public
   */
  packageManager: any

  /**
   * Install dependencies
   *
   * @param dependencies - Array of dependencies to install
   *
   * @public
   */
  install(
    dependencies: {
      name: string
      version: string
    }[],
  ): Promise<void>
}
