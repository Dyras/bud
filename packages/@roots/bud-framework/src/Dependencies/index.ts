import type {Service} from '../Service'

/**
 * @interface Dependencies
 */
export interface Dependencies extends Service {
  /**
   * Install production dependencies
   */
  install(
    dependencies: {
      name: string
      ver: string
      source: string
      type: 'dependencies' | 'devDependencies'
    }[],
  ): void
}
