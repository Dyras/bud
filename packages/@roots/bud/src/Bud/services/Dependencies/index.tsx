import {Service} from '@roots/bud-framework'
import {Ink, React} from '@roots/bud-support'
import {Dependencies as DependenciesManager} from '@roots/dependencies'
import {boundMethod as bind} from 'autobind-decorator'
import {readJsonSync} from 'fs-extra'
import {isEqual} from 'lodash'

class Dependencies extends Service<null> {
  public name = 'dependencies'

  /**
   * Interfaces with package manager
   */
  public manager: DependenciesManager

  @bind
  public register() {
    this.manager = new DependenciesManager(
      this.app.path('project'),
    )
  }

  @bind
  public readProjectJson() {
    return readJsonSync(this.app.path('project', 'package.json'))
  }

  @bind
  public overrideInstallTarget(
    dep: string,
    proposed: string,
  ): boolean {
    const pkgJson = this.readProjectJson()
    const checkAgainst =
      proposed == 'dependencies'
        ? 'devDependencies'
        : 'dependencies'

    if (Object.keys(pkgJson[checkAgainst] ?? {}).includes(dep)) {
      return true
    }

    return false
  }

  @bind
  public install(
    deps: {
      name: string
      ver: string
      source: string
      type: 'devDependencies' | 'dependencies'
    }[],
  ): void {
    this.app.dashboard.render(
      `Installing required peer dependencies`,
      `Installing`,
    )

    /**
     * Filter out ineligible packages
     */
    const installed = deps.map(dep => {
      this.app.dashboard.render(
        <>
          <Ink.Box marginBottom={1} flexDirection="column">
            <Ink.Text
              backgroundColor={this.app.store.get(
                'theme.colors.primary',
              )}
              color={this.app.store.get(
                'theme.colors.foreground',
              )}>
              Installing
            </Ink.Text>
          </Ink.Box>

          <Ink.Box marginBottom={1} flexDirection="column">
            <Ink.Text
              key={`${dep.name}-${dep.ver}`}>{`Installing ${dep.name}@${dep.ver}`}</Ink.Text>
          </Ink.Box>
        </>,
      )

      this.manager.client.install(
        isEqual(dep.type, 'devDependencies') &&
          !this.overrideInstallTarget(dep.name, dep.type),
        `${dep.name}@${dep.ver}`,
      )

      return dep
    })

    this.app.dashboard.render(
      <Ink.Box flexDirection="column">
        <Ink.Box marginBottom={1} flexDirection="column">
          <Ink.Text
            backgroundColor={this.app.store.get(
              'theme.colors.primary',
            )}
            color={this.app.store.get(
              'theme.colors.foreground',
            )}>
            Installation Complete
          </Ink.Text>
        </Ink.Box>

        {installed.length > 0 && (
          <Ink.Box marginBottom={1} flexDirection="column">
            {installed.map(dep => (
              <Ink.Text
                key={`${dep.name}-${dep.ver}`}>{`âœ“ ${dep.name}@${dep.ver}`}</Ink.Text>
            ))}
          </Ink.Box>
        )}
      </Ink.Box>,
    )
  }
}

export {Dependencies}
