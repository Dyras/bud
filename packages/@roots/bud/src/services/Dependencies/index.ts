import * as Framework from '@roots/bud-framework'

import {
  $,
  bind,
  DependenciesManager,
} from './dependencies.dependencies'

/**
 * Bud Dependencies Service class
 *
 * @public
 */
export class Dependencies
  extends Framework.Service
  implements Framework.Dependencies
{
  /**
   * Service ident
   *
   * @public
   */
  public ident = 'bud.dependencies'

  /**
   * Dependencies installation manager
   *
   * @public
   */
  public manager: DependenciesManager

  /**
   * Record of installed packages
   *
   * @public
   */
  public installed: Array<[string, string]> = []

  /**
   * {@link @roots/bud-framework#Service.register}
   *
   * @public
   * @decorator `@bind`
   */
  @bind
  public async register(): Promise<void> {
    this.manager = new DependenciesManager(
      this.app.path('project'),
    )
  }

  /**
   * Installs all the things
   *
   * @internalRemarks
   * #TODO: Fix this mess of a function and make it better. It's not good. -- GPT3
   *
   * @param deps - dependencies to install
   *
   * @public
   * @decorator `@bind`
   */
  @bind
  public async install(
    dependencies: {
      name: string
      version: string
    }[],
  ): Promise<void> {
    const install = dependencies.reduce(
      (acc, dependency) =>
        `${acc} ${dependency.name}@${dependency.version}`,
      ``,
    )
    const pkgManager = this.manager.isYarn() ? `yarn` : `npm`
    const command = `${pkgManager} add ${install} --dev`

    this.log(`await`, {
      prefix: 'install',
      message: install,
    })

    const installation = $(command)
    installation.stderr.pipe(process.stderr)

    await installation

    this.log(`success`, {
      prefix: 'install',
      message: install,
    })
  }
}
