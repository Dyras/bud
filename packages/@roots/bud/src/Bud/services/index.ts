import {Framework} from '@roots/bud-framework'

import {Api} from './Api'
import {Build} from './Build'
import {Cache} from './Cache'
import {Compiler} from './Compiler'
import {Dashboard} from './Dashboard'
import {Dependencies} from './Dependencies'
import {Env} from './Env'
import {Extensions} from './Extensions'
import {Hooks} from './Hooks'
import {Logger} from './Logger'
import {Project} from './Project'
import {Server} from './Server'

const services: Framework.Services = {
  api: Api,
  build: Build,
  cache: Cache,
  compiler: Compiler,
  dashboard: Dashboard,
  dependencies: Dependencies,
  project: Project,
  env: Env,
  extensions: Extensions,
  hooks: Hooks,
  logger: Logger,
  server: Server,
}

export {services}
