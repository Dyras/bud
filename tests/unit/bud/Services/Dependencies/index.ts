import {factory, Framework} from '@roots/bud'
import {Dependencies} from '@roots/bud/src/Bud/services/Dependencies'
import {Dependencies as DependenciesBase} from '@roots/dependencies'

describe('bud.project', function () {
  let bud: Framework
  let dependencies: Dependencies

  beforeAll(async () => {
    bud = factory()

    dependencies = new Dependencies(bud)
    dependencies.register()
  })

  afterAll(done => {
    bud.close(done)
  })

  it('has dependencies name', () => {
    expect(dependencies.name).toBe('dependencies')
  })

  it('has a register method', () => {
    expect(dependencies.manager).toBeInstanceOf(DependenciesBase)
  })

  it('has a readProjectJson method', () => {
    expect(dependencies.readProjectJson).toBeInstanceOf(Function)
  })

  it('readProjectJson', () => {
    expect(dependencies.readProjectJson().name).toMatchSnapshot()
  })

  it('has a shouldInstall method', () => {
    expect(dependencies.shouldInstall).toBeInstanceOf(Function)
  })

  it('has a install method', () => {
    expect(dependencies.install).toBeInstanceOf(Function)
  })
})
