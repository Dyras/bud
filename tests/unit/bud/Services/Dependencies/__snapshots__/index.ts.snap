// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bud.project readProjectJson 1`] = `
Object {
  "dependencies": Object {
    "tslib": "^2.3.1",
  },
  "description": "‚ö°Ô∏è Lightning fast frontend build tools combining the best parts of Symfony Encore and Laravel Mix",
  "devDependencies": Object {
    "@skypack/package-check": "0.2.2",
    "@svgr/webpack": "5.5.0",
    "@types/fs-extra": "9.0.12",
    "@types/jest": "^27.0.1",
    "@types/lodash": "4.14.172",
    "@types/node": "16.6.1",
    "@types/react": "17.0.19",
    "@types/react-dom": "17.0.9",
    "@types/react-helmet": "6.1.2",
    "@types/react-reconciler": "0.26.3",
    "@types/react-router-dom": "5.1.8",
    "@types/signale": "1.4.2",
    "@types/source-map-support": "^0",
    "@types/weak-napi": "2.0.0",
    "@typescript-eslint/eslint-plugin": "4.29.3",
    "@typescript-eslint/parser": "4.29.3",
    "@vercel/ncc": "0.30.0",
    "@yarnpkg/builder": "3.0.1",
    "autobind-decorator": "2.4.0",
    "codecov": "3.8.3",
    "dotenv": "10.0.0",
    "eslint": "7.32.0",
    "eslint-config-prettier": "8.3.0",
    "eslint-plugin-import": "2.24.2",
    "eslint-plugin-prettier": "3.4.1",
    "eslint-plugin-react": "7.25.0",
    "eslint-plugin-react-hooks": "4.2.0",
    "eslint-plugin-simple-import-sort": "7.0.0",
    "execa": "5.1.1",
    "file-loader": "6.2.0",
    "fs-extra": "10.0.0",
    "jest": "27.1.0",
    "lodash": "4.17.21",
    "prettier": "2.3.2",
    "raw-loader": "4.0.2",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-reconciler": "0.26.2",
    "rimraf": "3.0.2",
    "signale": "1.4.0",
    "source-map-support": "^0.5.19",
    "ts-jest": "27.0.5",
    "ts-node": "10.2.1",
    "tsconfig-paths": "3.11.0",
    "type-fest": "2.1.0",
    "typescript": "4.4.2",
    "url-loader": "4.1.1",
    "weak-napi": "2.0.2",
  },
  "engines": Object {
    "node": "^16.3.0",
  },
  "license": "MIT",
  "manifest": Object {
    "contributors": Object {
      "catgofire": Object {
        "type": Array [
          "‚úçüèΩ docs",
        ],
      },
      "clayrisser": Object {
        "type": Array [
          "‚úçüèΩ docs",
        ],
      },
      "kellymears": Object {
        "type": Array [
          "üöÄ steward",
          "üíª dev",
          "‚úçüèΩ docs",
        ],
      },
      "qwp6t": Object {
        "type": Array [
          "üöÄ steward",
          "üíª dev",
          "‚úçüèΩ docs",
        ],
      },
      "swalkinshaw": Object {
        "type": Array [
          "üöÄ steward",
          "‚úçüèΩ docs",
        ],
      },
    },
    "core": Object {
      "@roots/bud": Object {
        "dependencies": Array [
          "@roots/bud-api",
          "@roots/bud-build",
          "@roots/bud-cache",
          "@roots/bud-compiler",
          "@roots/bud-dashboard",
          "@roots/bud-extensions",
          "@roots/bud-framework",
          "@roots/bud-hooks",
          "@roots/bud-server",
          "@roots/bud-support",
        ],
        "description": "A frontend build tooling framework combining the best parts of Symfony Encore and Laravel Mix",
      },
      "@roots/bud-api": Object {
        "dependencies": Array [
          "@roots/bud-framework",
          "@roots/bud-support",
        ],
        "description": "@roots/bud core: build config facades",
      },
      "@roots/bud-build": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
        "description": "@roots/bud core: build config controller",
      },
      "@roots/bud-cache": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
        "description": "@roots/bud core: build config caching",
      },
      "@roots/bud-compiler": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-dashboard": Object {
        "dependencies": Array [
          "@roots/bud-framework",
          "@roots/bud-support",
        ],
      },
      "@roots/bud-extensions": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-framework": Object {
        "dependencies": Array [
          "@roots/container",
          "@roots/dependencies",
        ],
      },
      "@roots/bud-hooks": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-server": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-support": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
    },
    "deprecated": Array [
      "@roots/bud-cli",
      "@roots/bud-typings",
    ],
    "description": "‚ö°Ô∏è Lightning fast frontend build tools combining the best parts of Symfony Encore and Laravel Mix",
    "extensions": Object {
      "@roots/bud-babel": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
        "description": "Babel support for @roots/bud projects",
      },
      "@roots/bud-compress": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-criticalcss": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-emotion": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-entrypoints": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-esbuild": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-eslint": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-imagemin": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-library": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-mdx": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-postcss": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-preset-recommend": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-prettier": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-purgecss": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-react": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-sass": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-solid": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-stylelint": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-tailwindcss": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-terser": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-typescript": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-vue": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-wordpress-dependencies": Object {
        "dependencies": Array [
          "@roots/bud-framework",
          "@roots/merged-manifest-webpack-plugin",
        ],
      },
      "@roots/bud-wordpress-externals": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/bud-wordpress-manifests": Object {
        "dependencies": Array [
          "@roots/bud-framework",
        ],
      },
      "@roots/sage": Object {
        "dependencies": Array [
          "@roots/bud",
          "@roots/bud-babel",
          "@roots/bud-entrypoints",
          "@roots/bud-eslint",
          "@roots/bud-postcss",
          "@roots/bud-prettier",
          "@roots/bud-react",
          "@roots/bud-stylelint",
          "@roots/bud-tailwindcss",
          "@roots/bud-wordpress-dependencies",
          "@roots/bud-wordpress-externals",
          "@roots/bud-wordpress-manifests",
        ],
      },
    },
    "lib": Object {
      "@roots/container": Object {
        "dependencies": Array [],
      },
      "@roots/critical-css-webpack-plugin": Object {
        "dependencies": Array [],
      },
      "@roots/dependencies": Object {
        "dependencies": Array [],
      },
      "@roots/entrypoints-webpack-plugin": Object {
        "dependencies": Array [],
      },
      "@roots/filesystem": Object {
        "dependencies": Array [],
      },
      "@roots/ink-prettier": Object {
        "dependencies": Array [],
      },
    },
    "logo": "https://cdn.roots.io/app/uploads/logo-bud.svg",
    "name": "bud.js",
    "organization": Object {
      "favicon": "https://roots.io/favicon.ico",
      "name": "Roots Foundation, LLC",
      "site": "https://roots.io",
      "twitter": "https://twitter.com/roots",
    },
    "sponsors": Array [
      Object {
        "image": "https://cdn.roots.io/app/uploads/kinsta.svg",
        "title": "Kinsta",
        "url": "https://kinsta.com/?kaid=OFDHAJIXUDIV",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/km-digital.svg",
        "title": "KM Digital",
        "url": "https://k-m.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/carrot.svg",
        "title": "Carrot",
        "url": "https://carrot.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/c21redwood.svg",
        "title": "C21 Redwood Realty",
        "url": "https://www.c21redwood.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/wordpress.svg",
        "title": "WordPress.com",
        "url": "https://wordpress.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/icons8.svg",
        "title": "Icons8",
        "url": "https://icons8.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/harness-software.svg",
        "title": "Harness Software",
        "url": "https://www.harnessup.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/coders-clan.svg",
        "title": "Coders Clan",
        "url": "https://www.codersclan.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/genero.svg",
        "title": "Genero",
        "url": "https://generodigital.com/",
      },
      Object {
        "image": "https://cdn.roots.io/app/uploads/motto.svg",
        "title": "Motto",
        "url": "https://motto.ca/roots",
      },
    ],
    "url": Object {
      "discourse": "https://discourse.roots.io/c/bud/24",
      "docs": "https://budjs.netlify.app",
      "git": "git@github.com:roots/bud",
      "web": "https://github.com/roots/bud",
    },
    "version": Object {
      "increment": 12,
      "major": 4,
      "minor": 7,
      "patch": 0,
    },
  },
  "name": "bud.js",
  "packageManager": "yarn@3.0.1",
  "private": true,
  "repository": Object {
    "type": "git",
    "url": "https://github.com/roots/bud.git",
  },
  "resolutions": Object {
    "@oclif/config@1.17.0": "patch:@oclif/config@npm:1.17.0#.yarn/patches/@oclif-config-npm-1.17.0-acbde241aa.patch",
    "@types/react": "17.0.19",
    "ink": "3.0.9",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "typescript": "4.4.2",
    "yoga-layout-prebuilt": "patch:yoga-layout-prebuilt@*#.yarn/patches/yoga-layout-prebuilt.patch",
  },
  "version": "4.7.0-next.11",
  "volta": Object {
    "node": "16.3.0",
  },
  "workspaces": Object {
    "packages": Array [
      "examples/*",
      "packages/@roots/*",
      "packages/base",
      "site",
      ".yarn/local",
    ],
  },
}
`;
